src/django-app/quotes/policies.py

```diff
@@ -0,0 +1,136 @@
+from rest_access_policy import AccessPolicy
+from accounts.models import EmployeeData
+from shops.models import Shop
+from .models import QuoteRequest
+
+
+class QuoteAccessPolicy(AccessPolicy):
+    statements = [
+        {
+            "action": ["list"],
+            "principal": "authenticated",
+            "effect": "allow",
+        },
+        {
+            "action": ["retrieve"],
+            "principal": "authenticated",
+            "effect": "allow",
+        },
+        {
+            "action": ["create"],
+            "principal": "authenticated",
+            "effect": "allow",
+            "condition": ["has_quote_create_permission"],
+        },
+        {
+            "action": ["partial_update", "destroy"],
+            "principal": "authenticated",
+            "effect": "allow",
+            "condition": ["is_shop_related"],
+        },
+        {
+            "action": ["update"],
+            "principal": "*",
+            "effect": "deny",
+        },
+    ]
+
+    @classmethod
+    def scope_queryset(cls, request, qs):
+        if request.user.type == "customer":
+            return qs.filter(quote_request__user=request.user)
+        elif request.user.type == "shop_owner":
+            return qs.filter(shop__shop_owner=request.user)
+        elif request.user.type == "employee":
+            employee_shop = EmployeeData.objects.get(user=request.user).shop
+            return qs.filter(shop=employee_shop)
+        return qs
+
+    @classmethod
+    def scope_fields(cls, request, fields: dict, instance=None):
+        # TODO: This breaks the docs! Will leave this out.
```

Comment:
If you meant to leave this in, that's fine, but you've commented to leave this bit of the code out, so I'm pointing it out incase you forgot to remove it.

Reply:
Yeah, my comment is a bit misleading. I have left out the feature (it is not scoping out fields currently) but I have left in the code for it so I can fix it in the future and implement it.